src/aiida/tools/query/formatting.py

```diff
@@ -48,10 +53,25 @@ def format_state(process_state: str, paused: bool | None = None, exit_status: in
         # Unknown process state, use invisible separator
         symbol = '\u00b7'  # middle dot
 
+    if coloring:
+        if process_state in ['created', 'running']:
+            color = 'blue'
+        elif process_state in ['waiting']:
+            color = 'yellow'
+        elif process_state in ['finished']:
+            color = 'green'
+        elif process_state in ['killed', 'excepted']:
+            color = 'red'
+        else:
+            color = None
+    else:
+        color = None
```

Comment:
there was the option to put the coloring logic into `format_process_state`, but since this part already handles the symbol logic, it made more sense to me to put it here for readability.